### DISCUSSION TITLE
In tiling mode, where should new windows appear?

### DISCUSSION THREAD

**Discussion by leviport** [pop-os/shell#1225] [Ideas]
When in tiling mode, new windows used to open on the display that the mouse cursor was on. This behavior was changed with https://github.com/pop-os/shell/commit/49aba222933d8b0ed0a1a1cb7e82e366db0c1551 so that new windows _always_ branch from the currently active window, no matter which display the mouse cursor is on.

This can lead to some jarring behavior like windows appearing on a different display from the dock or launcher they were launched from, as shown in the recording below: 

https://user-images.githubusercontent.com/13512097/138780096-485994b4-e401-4cca-86e8-26e3517b63d3.mp4

I am having a hard time deciding whether the old way or the new way makes the most sense, so here's my idea: what if we made it a setting? I think the options could be something along the lines of, "New windows branch from active window" or "New windows follow mouse".

**Comment by mmstick** [comment]
Windows should open on the display where the mouse cursor is, as that is the active display. Then they should tile onto the active window on that display, if any.

**Reply by jacobgkau** [reply]
What happens if the mouse cursor is on a display with several windows, none of them active? Which window on that display does the new window attach to in that case?

**Reply by mmstick** [reply]
I think the current behavior is to choose the largest window on the display to attach to. We could of course automatically assign focus to the last-focused window on that particular display if the last-focused window is still around.

**Reply by leviport** [reply]
Phew, I thought current behavior was the intended behavior. I'll create an issue from this discussion then.

**Reply by jacobgkau** [reply]
> Phew, I thought current behavior was the intended behavior. I'll create an issue from this discussion then.

The current behavior was called intended in https://github.com/pop-os/shell/pull/1171, but I think we're all okay with changing it back/defining it better.

**Reply by Hyperspeed1313** [reply]
Is this thread still in work? I've just moved to Nobara and am trying out the Pop Shell extension and it still behaves as described in #1171.

Seeing as there are multiple ways to launch a new window, I think the best solution to fit most people's need would be an adaptive solution based on the input device that spawned the window, if such a solution is possible: If the last input was via mouse or touchscreen, it opens on the monitor that was just interacted with. But if the last input was a keyboard then it follows the current behavior of tiling off the active window.

**Reply by mmstick** [reply]
There's no more feature development on pop-shell. I'm working on COSMIC now.
